src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -42,7 +42,7 @@ namespace {
 static DnssdCache<CHIP_CONFIG_MDNS_CACHE_SIZE> sDnssdCache;
 #endif
 
-static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR error)
+static void HandleNodeResolve(void * context, const Span<DnssdService> & results, CHIP_ERROR error)
```

Comment:
How about allowing multiple calls of HandleNodeResolve instead of just one?
That way we do not need special interface handling.

Reply:
"Allowing" in the sense of not doing the Release on the proxy?

The issue is that the consumers treat "OnNodeDiscoveryComplete", which is the API they get to work with, as meaning "complete", so they would end up ignoring all the results after the first one anyway, no?